src/find.ts

```diff
@@ -25,8 +30,11 @@ export async function find(filename: string) {
 	throw new Error(`no tsconfig file found for ${filename}`);
 }
 
-async function tsconfigInDir(dir: string): Promise<string | void> {
+async function tsconfigInDir(dir: string, options?: TSConfckFindOptions): Promise<string | void> {
 	const tsconfig = path.join(dir, 'tsconfig.json');
+	if (options?.tsConfigPaths?.has(tsconfig)) {
+		return tsconfig;
+	}
```

Comment:
```suggestion
	if (options?.tsConfigPaths) {
		return options.tsConfigPaths.has(tsconfig) ? tsconfig : undefined;
	}
```
I think this is better.
Since `tsConfigPaths` has all tsconfig paths inside root, tsconfig does not exist if it is not included in `tsConfigPaths`.

Reply:
you're right. fixed